patches/api/0458-Make-spawn-egg-color-visible.patch

```diff
@@ -0,0 +1,28 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Luis <luisc99@icloud.com>
+Date: Thu, 11 Jan 2024 19:58:17 +0100
+Subject: [PATCH] Make spawn egg color visible
+
+
+diff --git a/src/main/java/org/bukkit/UnsafeValues.java b/src/main/java/org/bukkit/UnsafeValues.java
+index a4b38f284d4fea7df7f9df9bf44e4f68fefaf20f..5eb77ce44898f16379e92d53420df8f72ea5b591 100644
+--- a/src/main/java/org/bukkit/UnsafeValues.java
++++ b/src/main/java/org/bukkit/UnsafeValues.java
+@@ -260,4 +260,17 @@ public interface UnsafeValues {
+ 
+     String getStatisticCriteriaKey(@NotNull org.bukkit.Statistic statistic);
+     // Paper end
++
++    // Paper start - spawn egg color visibility
++    /**
++     * Obtains the underlying color informating for a spawn egg of a given
++     * entity type, or -1 if the entity passed does not have a spawn egg.
++     * Spawn eggs have two colors - the background layer (0), and the
++     * foreground layer (1)
++     * @param entityType The entity type to get the color for
++     * @param layer The texture layer to get a color for
++     * @return The color of the layer for the entity's spawn egg
++     */
++    int getSpawnEggLayerColor(org.bukkit.entity.EntityType entityType, int layer);
```

Comment:
Does the int represent a RGB or RGBA color value or is it just a index for an enum Color?
Please either clarify in JavaDoc or return a `@Nullable Color` instead.

Reply:
Yeah it should be a Color object, not the raw int